plugins/woocommerce/src/StoreApi/Utilities/OrderController.php

```diff
@@ -171,6 +171,41 @@ public function validate_order_before_payment( \WC_Order $order ) {
 		$this->validate_email( $order );
 		$this->validate_selected_shipping_methods( $needs_shipping, $chosen_shipping_methods );
 		$this->validate_addresses( $order );
+
+		// Perform custom validations.
+		$this->perform_custom_order_validation( $order );
+	}
+
+	/**
+	 * Perform custom order validation via WooCommerce hooks.
+	 *
+	 * Allows plugins to perform custom validation before payment.
+	 *
+	 * @param \WC_Order $order Order object.
+	 * @throws RouteException Exception if validation fails.
+	 */
+	protected function perform_custom_order_validation( \WC_Order $order ) {
+		$validation_errors = new \WP_Error();
+
+		/**
+		 * Allow plugins to perform custom validation before payment.
+		 *
+		 * Plugins can add errors to the $validation_errors object.
+		 *
+		 * @param \WC_Order $order             The order object.
+		 * @param \WP_Error $validation_errors WP_Error object to add custom errors to.
+		 * @since 9.5.0
```

Comment:
This should be changed to 9.9.0 since we missed the feature freeze for 9.8.0

```suggestion
		 * @since 9.9.0
```